src/datachain/lib/file.py

```diff
@@ -486,13 +504,223 @@ class ImageFile(File):
     def read(self):
         """Returns `PIL.Image.Image` object."""
         fobj = super().read()
-        return Image.open(BytesIO(fobj))
+        return PilImage.open(BytesIO(fobj))
 
     def save(self, destination: str):
         """Writes it's content to destination"""
         self.read().save(destination)
 
 
+class Image(DataModel):
+    """`DataModel` for image file meta information."""
+
+    width: int = Field(default=-1)
+    height: int = Field(default=-1)
+    format: str = Field(default="")
+
+
+class VideoFile(File):
+    """`DataModel` for reading video files."""
+
+    def get_info(self) -> "Video":
+        """Returns video file information."""
+        from .video import video_info
+
+        return video_info(self)
+
+    def get_frame_np(self, frame: int) -> "ndarray":
```

Comment:
thinking out-loud here but should a frame be an `ImageFile` and `ImageFile` have a `to_ndarray` method?

Reply:
I'll take a look at the notebook but my thought would be that you would want to be able to call something like `video.split_to_frame` with an optional start/end frame + optional destination path and DataChain would split the video into frames and upload them all to a bucket as images.

Reply:
The frames use-case could end up looking something like:

(
    DataChain.from_storage("gs://datachain-demo/some-desc/videos")
    .limit(20)
    .gen(frame=file.split_to_frame, params="file", output={"frame": ImageFile})
    .setup(yolo=lambda: YOLO("yolo11n.pt"))
    .map(boxes=process_bboxes)
    .show()
)

Reply:
This also can be done by the `save_frames` method below, or we can add new method `upload_frames` to upload images to the storage instead of saving them.

Reply:
`save_frames` without upload breaks the promise of dataset reproducibility